modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIEndpointRelevantObjectEntryModelListener.java

```diff
@@ -151,16 +150,43 @@ private void _validate(ObjectEntry objectEntry) {
 					apiApplicationId, responseAPISchemaId, scope);
 			}
 
+			String pathString = (String)values.get("path");
+
+			Http.Method method = Http.Method.valueOf(
+				StringUtil.toUpperCase((String)values.get("httpMethod")));
+
 			if (Objects.equals(
 					APIApplication.Endpoint.RetrieveType.parse(
 						(String)values.get("retrieveType")),
 					APIApplication.Endpoint.RetrieveType.SINGLE_ELEMENT)) {
 
+				String pathInParameterString = StringUtil.extractLast(
+					pathString, StringPool.FORWARD_SLASH);
+
 				String pathParameter = (String)values.get("pathParameter");
 
-				_validateSingleElementPath(
-					objectEntry, pathParameter, pathString,
-					responseAPISchemaId);
+				if (Objects.equals(method, Http.Method.POST)) {
```

Comment:
I think it could be simplified depending on the http method like:

```java
Http.HttpMethod method =  (String)values.get("httpMethod"); 
if(method =="post"){

     _validatePostEndpoint()

}else if(method =="get"){

    _validateGetEndpoint()

}
``` 
I think that it could help for next http methods that we will add in the future as well as it will increase the complexity of analyzing the validations. What do you think?